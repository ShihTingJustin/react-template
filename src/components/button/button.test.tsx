import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Button from '.';

describe('Test Button component', () => {
  const component = (
    <Button onClick={() => {}}>
     <div>button text</div>
    </Button>
  );

  it('[Render] match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('[Render] toBeInTheDocument', () => {
    render(component);
    const element = screen.getByTestId('button');
    expect(element).toBeInTheDocument();
  });
});
