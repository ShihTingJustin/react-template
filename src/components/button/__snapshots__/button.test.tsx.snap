// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component [Render] match snapshot 1`] = `
<div
  className="button-root"
  data-testid="button"
>
  <button
    onClick={[Function]}
  >
    <div>
      button text
    </div>
  </button>
</div>
`;
